src/bot/commands/public/settings.command.ts

```diff
@@ -1,198 +1,43 @@
-        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-        // @ts-ignore
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/no-unsafe-member-access
-        context.menu.nav('settingsAirRaidAlertSubmenu');
-        await context.editMessageText(getAirRaidAlarmSettingsMessage(context.chatSession.chatSettings), { parse_mode: 'HTML' });
-      })
-      .text(airAlarmNotificationMessage, isAlarmNow, isStateSelected, async (context) => {
-        context.chatSession.chatSettings.airRaidAlertSettings.notificationMessage =
-          !context.chatSession.chatSettings.airRaidAlertSettings.notificationMessage;
-        alarmChatService.updateChat(context.chatSession, context.chat?.id);
-        const newText = getSettingsMenuMessage(context.chatSession.chatSettings);
+      if (isChatPrivate) {
+        const userSession = await redisService.getUserSession(userId);
+        const chats = userSession?.linkedChats || [];
+        return chats.length > 0
+          ? context.reply(linkToWebView)
+          : context.reply(`You don't linked chats yet. Please, go in group and press /settings.`);
+      }
 
-        if (context.msg?.text !== newText) {
-          await context.editMessageText(newText, { parse_mode: 'HTML' });
+      if (!isChatPrivate) {
+        const admins = await context.api.getChatAdministrators(chatId);
+        if (!admins.some((object) => object.user.id.toString() === userId)) {
+          return context.reply(`You are not admin`);
         }
-      })
-      .text(turnOffChatWhileAlarmButton, isAlarmNow, isStateSelected, async (context) => {
-        await toggleSetting(context, 'disableChatWhileAirRaidAlert');
-        alarmChatService.updateChat(context.chatSession, context.chat?.id);
-      })
-      // TODO UABOT-2 COMMENT UNTIL DESCRIPTION WILL BE AVAILABLE
-      // .row()
-      // .submenu(settingsDescriptionButton, 'settingsDescriptionSubmenu', (ctx) => {
-      //   ctx.editMessageText(detailedSettingsDescription).catch(handleError);
-      // })
-      .row()
-      .text(settingsSubmitMessage, async (context) => {
-        await context.answerCallbackQuery({
-          text: settingsSet,
-          show_alert: true,
-        });
-        await context.deleteMessage();
-      });
-
-    return this.settingsMenuObj;
-  }
 
-  initAirRaidAlertSubmenu() {
-    this.settingsAirRaidAlertObj = new MiddlewareMenu('settingsAirRaidAlertSubmenu')
-      .addGlobalMiddlewares(onlyAdmin)
-      .dynamic((context, range) => dynamicLocationMenu(context, range, this.airRaidAlarmStates))
-      .row()
-      .back(goBackButton, (context) =>
-        context.editMessageText(getSettingsMenuMessage(context.chatSession.chatSettings), { parse_mode: 'HTML' }),
-      );
+        const userSession = await redisService.getUserSession(userId);
+        const chats = userSession?.linkedChats || [];
+        const chatIndex = chats.findIndex((chat) => chat.id.toString() === chatId);
```

Comment:
You used the `.some` method before.
It's better to use `.some` here as well.
Rename it to `isChatExists` or `isChatMissing` (if you flip the equation)